test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightDeltaWriterTests.cs

```diff
@@ -1434,7 +1434,7 @@ public void WriteContentIn41DeletedEntry(bool isResponse)
         [InlineData(/*isResponse*/true)]
         [InlineData(/*isResponse*/false)]
         [Theory]
-        public void WriteDeletedEntryWithoutKeyOrIdShouldFail(bool isResponse)
+        public void WriteDeletedEntryWithoutKeyOrIdShouldNotFail(bool isResponse)

```

Comment:
Won't this be a breaking change?

From the issue, have you considered this?
`May be able to leverage transient entities (i.e., requiring setting IsTransient=true when writing the resource).`